pkg/groups/service.go

```diff
@@ -679,10 +679,6 @@ func (s *V1Service) GetGroupIdentities(ctx context.Context, groupId string, para
 	ctx, span := s.tracer.Start(ctx, "groups.V1Service.GetGroupIdentities")
 	defer span.End()
 
-	if principal := authentication.PrincipalFromContext(ctx); principal == nil {
-		return nil, v1.NewAuthorizationError("unauthorized")
-	}
-
```

Comment:
Tagging @nsklikas since we added this pieces of code following one of his comments. This was done to be extra safe.
I am fine removing it, but I think we should agree since afaik there's diverging choices on this particular subject.